src/buildercore/bootstrap.py

```diff
@@ -486,7 +486,15 @@ def _update_ec2_node():
         # write out environment config (/etc/cfn-info.json) so Salt can read CFN outputs
         write_environment_info(stackname, overwrite=True)
 
+        # bit of a hack, but project config merging doesn't apply to top-level values
+        # in this case we look for an alternate salt version under a project's 'ec2' section
         salt_version = context['project']['salt']
+        alt_salt_version = context['ec2'].get('salt')
```

Comment:
eventually it will be correct for `salt` to be under `ec2` as it's not used without EC2 servers involved (e.g. the `generic-cdn` project). Would support that change, in a separate PR

Reply:
yes, I started going down that route but it got complex quickly. Another option might be to revisit the configuration merging and get rid of aws-alt, gcp-alt, vagrant-alt and simply have 'alt' configs where any part of the structure could be overridden. That's a bit more ambitious though ;)